app/Actions/Photo/Toggles.php

```diff
@@ -14,21 +12,15 @@
  */
 class Toggles
 {
-	public $property;
+	public string $property;
 
 	public function do(array $photoIDs): bool
 	{
-		try {
-			//! DB::raw is safe because WE (dev) have control over $property. It is not influced by user inputs.
-			$no_error = Photo::whereIn('id', $photoIDs)->update([$this->property => DB::raw('1 XOR `' . $this->property . '`')]);
-		} catch (QueryException $e) {
-			// for Sqlite we need the slow approach
-			$photos = Photo::whereIn('id', $photoIDs)->get();
-			$no_error = true;
-			foreach ($photos as $photo) {
-				$photo->{$this->property} = $photo->{$this->property} != 1 ? 1 : 0;
-				$no_error &= $photo->save();
-			}
+		$photos = Photo::query()->whereIn('id', $photoIDs)->get();
```

Comment:
Why using the slow approach and forcing query of multiple models when this could be done in a single query ?
Or does the XOR not work anymore?

Reply:
The main reasons were:

 1. Consistent coding approach with SQLite. What is the purpose of using an ORM layer if we write "bare metal" queries for each RDMS anyway? Also I found the idea to make throwing an exception the "normal" case very unpleasant.
 2. Using proper models also updates the `updated_at` column on `save`. That was actually the main reason.